File: src/FC.cpp
#include "FC.hpp"

int main(int argc, char **argv)
{
    char input;
    string line;
    fstream conf, dirstxt;
    // vector<string> dirs;
    conf.open("fc.conf", std::ios::in | std::ios::out);
    dirstxt.open("dirs.txt", std::ios::in | std::ios::app);
    if (!conf.is_open())
    {
        printf("Error opening config file\n");
        exit(EXIT_FAILURE);
    }

    if (!dirstxt.is_open())
    {
        printf("Error opening dirs.txt\n");
        exit(EXIT_FAILURE);
    }

    //Evaluate cmd args
    if (strcmp(argv[1], "add") == 0)
    {
        if (!strcmp(argv[2], "") == 0) //Not empty
        {
            cout << argv[2] << " added";
            dirstxt << argv[2] << endl;
        }
    }

    if (strcmp(argv[1], "ls") == 0)
    {
        int i = 1;
        while (!dirstxt.eof())
        {
            getline(dirstxt, line);
            if (!line.empty())
                cout << "Dir #" << i++ << ": " << line << endl;
        }
    }

    if (strcmp(argv[1], "clear") == 0)
    {
        printf("Clear dirs.txt? [Y/n] ");
        scanf("%c", &input);
        if (input != 'n')
        {
            dirstxt.close();
            dirstxt.open("dirs.txt", std::ios::out | std::ios::trunc);
            dirstxt.close();
        }
    }

    conf.close();
    dirstxt.close();
}

File: src/fc.py
from header import *

with open(os.environ["APPDATA"] + "/fc/fc.conf", 'r') as conf:
    for i, line in enumerate(conf):
        if "Default Copy Location:" in line:
            cpLoc = pathlib.PurePath(line[23:])  # Where folders are copied

with open(os.environ["APPDATA"] + "/fc/dirs.txt", 'r') as dirstxt:
    dirs = [pathlib.PurePath(line.rstrip()) for line in dirstxt]
    dirstxt.close()

if (len(sys.argv) == 1):
    copyDirs(dirs, cpLoc)
    exit()

elif (sys.argv[1] == "cp"):
    try:
        copyDirs(dirs, pathlib.PurePath(sys.argv[2]))
    except IndexError:
        copyDirs(dirs, cpLoc)
    exit()

elif (sys.argv[1] == "add"):
    try:
        for dir in dirs:  # Check for duplicates
            if str(sys.argv[2]) == str(dir):
                print(f"{colors.FAIL}dir '{str(sys.argv[2])}' is already in the file!{colors.ENDC}")
                listDirs(dirs)
                exit()

        dirs.append(sys.argv[2])
        with open(os.environ["APPDATA"] + "/fc/dirs.txt", 'w') as dirstxt:
            for dir in dirs:
                dirstxt.write(str(dir) + '\n')
    except IndexError:
        print("'add' requires a directory")
        exit()

elif (sys.argv[1] == "ls"):
    listDirs(dirs)
    exit()

elif (sys.argv[1] == "rm"):
    with open(os.environ["APPDATA"] + "/fc/dirs.txt", 'r') as f:
        lines = f.readlines()
    with open(os.environ["APPDATA"] + "/fc/dirs.txt", 'w') as f:
        for line in lines:
            if line.strip("\n") != sys.argv[2]:
                f.write(line)
    listDirs(dirs)
    exit()

elif (sys.argv[1] == "clr"):
    with open(os.environ["APPDATA"] + "/fc/dirs.txt", 'w') as dirstxt:
        dirstxt.write("")

elif (sys.argv[1] == "help"):
    help()
    exit()

else:
    print(f"{colors.FAIL}Unknown command: '{sys.argv[1]}'")
    help()
    exit()
